src/pages/workspaces/workspace/Data.js

```diff
@@ -732,7 +732,7 @@ export const WorkspaceData = _.flow(
       // setWdsProxyUrl({ status: 'Loading', state: '' })
       return Ajax()
         .Apps.listAppsV2(workspaceId)
-        .then(resolveWdsUrl)
+        .then((apps) => resolveWdsUrl(apps, /* raiseOnError= */ true))
```

Comment:
I didn't love adding `raiseOnError` because I think raising on error state should be the default behavior, but `resolveWdsUrl` was clearly implemented initially to return an empty string in an error state, so I think it was intentional.  I thought making this surgically enabled for this particular scenario would be safer than changing the behavior for all the clients.

Reply:
Hmm that seems weird.  It may have been because error vs. provisioning weren't well-distinguished and we didn't want to error while provisioning?  I would be in favor of having default behavior be to raise on error, or at least try and see if there's any sensible reason not to.

Reply:
Agreed on making raise on error the default (and only) behavior. I took a quick look at other places where `resolveWdsUrl` is used, and it looks like none of them handle the current "return empty string on error" behavior well.

Reply:
I think there's possibly some room for debate on what constitutes an error in this request. In some ways, it has successfully resolved the URL in this case. It's just that it's resolved that there is no URL.

This still leaves a gray area though... what about provisioning, etc states? They also don't return a proxy URL, so should they throw an error as well?

I suspect that `resolveWdsUrl` might not be a useful level of abstraction: it just doesn't return enough information for consumers to make a good decision on what to do. It might be preferable to use `resolveWdsApp` instead, since its response includes the app's state.

Reply:
Interesting points...I started by initially using `resolveWdsApp` and doing the logic inline, and didn't feel great about the duplication so I drove this flag down into `resolveWdsUrl`.  But I think you might be right about it being the wrong abstraction.  I guess the question is, do we dispense with `resolveWdsUrl` entirely since it isn't doing a great job at the moment?  I'll see what this looks like and put up a commit.

Reply:
Ok I took a look at what eliminating `resolveWdsUrl` would look like and I think it's too much to try to unwind right now.  So instead, I dropped the `raiseOnError` flag and just inlined the logic that I needed in the `Data.js` `loadWdsUrl` method.